### PR TITLE
Added the delete reminder function

### LINKED ISSUE #229: Delete Reminder Route
As a user, I need to have different endpoints that I can access to get data provided on the frontend. All these endpoints should be inside the `/src/controllers/reminder.controller.js` file

### Acceptance
- [ ] I have a `delete` endpoint that sends a `DELETE` request to the DB

### PR DESCRIPTION
**Pull request by MilitusInnocent**
This PR fixes #229 
[Live link](https://reminders.zuri.chat/api/v1/deletereminder)
A delete reminder function was added to the reminder controller file, and this function deletes a data using the id of the data from the DB by calling the zc_core database endpoint.
delete reminder function

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -13,5 +13,6 @@ router.get('/ping', (req, res) =>
 	res.json({ message: 'Hello! You have found the zc_plugin_reminder api' })
 )
 router.route('/reminders').get(reminderController.create)
+router.route('/reminders/:id').delete(reminderController.deleteById)
```
Comment (iyosayi) [inline src/routes/index.js]: What if they don't add the `ID`, how are we checking for this? Enforce error checks

**Review by iyosayi — CHANGES_REQUESTED** [review]
Great work, fix this tiny issue

File: src/routes/index.js (hunk 1 above)
Comment (MilitusInnocent) [reply src/routes/index.js]: I'll fix this right away.

**Comment by MilitusInnocent** [comment]
This PR fixes #229
[Live link](https://reminders.zuri.chat/api/v1/deletereminder)
A delete reminder function was added to the reminder controller file, and this function deletes a data using the id of the data from the DB by calling the zc_core database endpoint.


delete reminder fixed

File: src/controllers/reminder.controller.js [hunk 2]
```diff
@@ -8,6 +8,26 @@ const reminderController = {
...
 		return res.send('heelo')
 	},
+
+	deleteReminder: async (req, res) => {
+		try {
+			const db = makeDb()
+			const deleteReminder = await db.findByIdAndDelete(req.params.id)
+
+			if (!id) {
+				throw new Error('id is required')
```
Comment (iyosayi) [inline src/controllers/reminder.controller.js:140]: This should be the correct order

```javascript
if(!id) {}...
const db = makeDb()
```

**Review by iyosayi — CHANGES_REQUESTED** [review]
implement these

File: src/controllers/reminder.controller.js (hunk 2 above)
Comment (MilitusInnocent) [reply src/controllers/reminder.controller.js:140]: Hi King, this has been fixed.

**Comment by iyosayi** [comment]
@MilitusInnocent Please pull from dev branch into your branch then push

**Comment by MilitusInnocent** [comment]
> 
> 
> @MilitusInnocent Please pull from dev branch into your branch then push

Okay. I have. It says up to date. So I should push into the dev branch, then?

**Review by iyosayi — APPROVED** [review]

**Pull request merged**